autox/informer.go

```diff
@@ -3,134 +3,377 @@ package autox
+		log.Logger.Error("expected secret with label selector:", labelSelector, "but none were found")
+		return err
+	} else if secretsLen > 1 {
+		log.Logger.Error("expected secret with label selector:", labelSelector, "but more than one were found")
+		return err
+	}
+	secret := secretList.Items[0]
+
+	values := applicationValues{
+		name:      releaseName,
+		namespace: namespace,
+
+		owner: struct {
+			name string
+			uid  string
+		}{
+			name: secret.Name,
+			uid:  string(secret.GetUID()),
+		},
+
+		chart: struct {
+			url     string
+			name    string
+			values  map[string]interface{}
+			version string
+		}{
+			url:     releaseSpec.RepoURL,
+			name:    releaseSpec.Name,
+			values:  releaseSpec.Values,
+			version: releaseSpec.Version,
+		},
+	}
+
+	gvr := schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"}
+
+	tpl, err := base.CreateTemplate(tplStr)
+	if err != nil {
+		log.Logger.Error("could not create application template")
+		return err
+	}
+
+	var buf bytes.Buffer
+	err = tpl.Execute(&buf, values)
+	if err != nil {
+		log.Logger.Error("could not execute application template")
+		return err
+	}
 
-	// TODO: release Helm chart
+	log.Logger.Debug("application manifest:", buf.String())
+
+	// serialize application manifest into unstructured object
+	obj, _, err := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme).Decode(buf.Bytes(), nil, nil)
+	if err != nil {
+		log.Logger.Error("could not serialize application manifest")
+		return err
+	}
+	unstructuredMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		log.Logger.Error("could not convert application manifest into unstructured object")
+		return err
+	}
+	unstructuredObj := &unstructured.Unstructured{Object: unstructuredMap}
+
+	// TODO: what to put for ctx?
+	// create application object
+	_, err = k8sClient.dynamic().Resource(gvr).Namespace(namespace).Create(context.TODO(), unstructuredObj, metav1.CreateOptions{})
+	if err != nil {
+		log.Logger.Error("could not create application:", releaseName)
+		return err
+	}
 
 	log.Logger.Debug("Release chart:", releaseName)
+	return nil
 }
 
-func deleteHelmReleases(resourceName string) {
-	doChartAction(resourceName, deleteAction)
+// deleteHelmReleases for a given spec group
+func deleteHelmReleases(prunedLabels map[string]string, namespace string) error {
+	return doChartAction(prunedLabels, deleteAction, namespace)
 }
 
-func deleteHelmRelease(releaseName string, chart chart) {
-	// TODO: check if there is a preexisting Helm release
+// deleteHelmRelease with a given release name
+var deleteHelmRelease = func(releaseName string, group string, namespace string) error {
+	gvr := schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"}
 
-	// TODO: mutex
-
-	// TODO: delete Helm chart
+	err := k8sClient.dynamic().Resource(gvr).Namespace(namespace).Delete(context.TODO(), releaseName, metav1.DeleteOptions{})
```

Comment:
I'm confused by why delete helm release is implemented a delete of a `Deployment` object. Shouldn't we be deleting the `Application` object ?